Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -77,44 +75,38 @@ class ContactViewController: OWSViewController, ContactShareViewHelperDelegate {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
-        // Use "dark" style navigation bar in this view.
-        UIUtil.applyDefaultSystemAppearence()
+        // Use "dark" style status bar in this view,
+        // But we don't apply the entire default system UI (UIUtil.applyDefaultSystemAppearence) because
+        // doing so while pushing onto an existing nav controller who's navbars are hidden corrupts the
+        // title label/navbar items elsewhere in the stack.
+        CurrentAppContext().setStatusBarStyle(.default)
 
         guard let navigationController = self.navigationController else {
             owsFail("\(logTag) in \(#function) navigationController was unexpectedly nil")
             return
         }
         navigationController.isNavigationBarHidden = true
 
-        // self.navigationController is nil in viewDidDisappear when transition via message/call buttons
-        // so we maintain our own reference.
-        self.postDismissNavigationController = navigationController
```

Comment:
This was only needed if we're mutating the navigationController in view**Did**Disappear.